src/backends/tensorrt/plan_backend.cc

```diff
@@ -1329,16 +1355,25 @@ PlanBackend::Context::InitializeConfigExecuteOutputBindings(
       max_byte_size = std::max(max_byte_size, byte_size);
     }
 
-    // Allocate CUDA memory. We rely on buffer_bindings_ being non-nullptr to
-    // indicate that the buffer has been correctly initalized so even
-    // for zero-sized tensors always allocate something.
+    // Allocate CUDA memory. Use cudaHostAlloc if zero copy supported. We rely
+    // on buffer_bindings_ being non-nullptr to indicate that the buffer has
+    // been correctly initalized so even for zero-sized tensors always allocate
+    // something.
+    bool zero_copy_support = false;
     void* buffer;
-    cudaError_t err = cudaMalloc(&buffer, std::max((int64_t)1, max_byte_size));
+    cudaError_t err;
+    Status status = CheckGPUZeroCopySupport(gpu_device_, &zero_copy_support);
+    if (status.IsOk() && zero_copy_support) {
+      err = cudaHostAlloc(
+          &buffer, std::max((int64_t)1, max_byte_size), cudaHostAllocMapped);
```

Comment:
Where is 'buffer' freed? Same with other buffer init above. It needs to be freed with cudaHostFree, doesn't it?